### DISCUSSION TITLE
Masks (made on Flash) does now show right on the built-in 3d scenario

### DISCUSSION THREAD

**Discussion by leonardozimbres** [BlackMATov/unity-flash-tools#1] [Q&A] [answered]
Hi there! Dont know how to summarize but it's pretty simple: everything that is masked shows behind any 3d object on the sceneview/game.
Is there any way to solve this issue? This plugin is absolutely awesome.
Cheers!

**Comment by BlackMATov** [comment, marked as answer]
Hello and thank you :-)

Can you share a simple scene with the problem or explain in details about your issue?

Because I don't see any problem with masks:

<img width="1840" alt="Screenshot 2023-02-10 at 08 40 45" src="https://user-images.githubusercontent.com/2990282/217979247-b810b753-a3e5-46a7-80e3-45e2380ab208.png">

**Reply by leonardozimbres** [reply]
Hi BlackMatov! Its great to be in touch with you. I did tested a sample project, and the issue seems to happen when I change the rendering path from Forward to Deferred.
![image](https://user-images.githubusercontent.com/3225893/218140204-6c0570cd-accb-4fab-9b73-c7e4bed5e245.png)

**Reply by BlackMATov** [reply]
Oh yes, I see.  Unfortunately, deferred rendering is not currently supported. For masks and recursive mask, only forward rendering works well.

**Reply by BlackMATov** [reply]
But! You can use some trick. Set the [clearStencilAfterLightingPass](https://docs.unity3d.com/ScriptReference/Camera-clearStencilAfterLightingPass.html) property to `true` for all cameras you use (including the editor camera). Something like this:

```c#
foreach ( var camera in Camera.allCameras ) {
    camera.clearStencilAfterLightingPass = true;
}
```

This can help! :-)

**Reply by BlackMATov** [reply]
Can you try it in the [deferred-rendering-camera](https://github.com/BlackMATov/unity-flash-tools/tree/feature/deferred-rendering-camera) branch?

**Reply by leonardozimbres** [reply]
Hi BlackMator, it works indeed! It's very good to see it working, so good to see flash animations working on a Unity game, so smooth.
Thanks a lot!

**Reply by BlackMATov** [reply]
Merged to the main branch.
